cfme/tests/containers/test_basic_metrics.py

```diff
@@ -33,8 +40,10 @@ def enable_capacity_and_utilization(appliance):
 
     logger.info("Enabling metrics collection roles")
     appliance.server.settings.enable_server_roles(*args)
-
-    yield
+    if appliance.wait_for_server_roles(args):
```

Comment:
**REQUIRED**
Why do you need that? I have a test that enables these roles too, but I don't need to wait for them to be enabled https://github.com/ManageIQ/integration_tests/blob/master/cfme/tests/control/test_alerts.py#L197

Reply:
@quarckster I was seeing cases where the roles were not being set and the test would fail

Reply:
@jawatts can you please provide the steps how it can be reproduced?

Reply:
@quarckster What I observed was when my test ran ```appliance.server.settings.enable_server_roles(*args)```, it would fail on the navigation to the metrics page because the the field was disabled.

Reply:
@jawatts I commented on the wait_for_server_roles implementation.

This still won't catch any exceptions that occur if the role doesn't get enabled within the timeout, so the skip will never happen.